projects/storefrontlib/src/cms-components/product/carousel/product-references/product-references.component.service.spec.ts

```diff
@@ -0,0 +1,145 @@
+import { inject, TestBed } from '@angular/core/testing';
+import {
+  CmsProductReferencesComponent,
+  ProductReferenceService,
+  RoutingService,
+  UIProductReference,
+} from '@spartacus/core';
+import { Observable, of } from 'rxjs';
+import { CmsComponentData } from '../../../../cms-structure/page/model/cms-component-data';
+import { ProductReferencesService } from './product-references.component.service';
+
+const productCode = 'productCode';
+const product = {
+  code: productCode,
+  name: 'testProduct',
+};
+
+const title = 'mockTitle';
+
+const list: UIProductReference[] = [
+  { referenceType: 'SIMILAR', target: product },
+  { referenceType: 'ACCESSORIES', target: product },
+];
+
+const router = {
+  state: {
+    url: '/',
+    queryParams: {},
+    params: { productCode },
+  },
+};
+
+class MockRoutingService {
+  getRouterState(): Observable<any> {
+    return of(router);
+  }
+}
+
+class MockProductReferenceService {
+  get(): Observable<UIProductReference[]> {
+    return of(list);
+  }
+}
+
+const mockComponentData: CmsProductReferencesComponent = {
+  uid: 'MockProductReferencesComponent',
+  typeCode: 'ProductReferencesComponent',
+  title: 'mockTitle',
+  productReferenceTypes: 'referenceType',
+};
+
+const MockCmsComponentData = <CmsComponentData<any>>{
+  data$: of(mockComponentData),
+};
+
+describe('ProductReferencesService', () => {
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        {
+          provide: CmsComponentData,
+          useValue: MockCmsComponentData,
+        },
+        {
+          provide: ProductReferenceService,
+          useClass: MockProductReferenceService,
+        },
+        {
+          provide: RoutingService,
+          useClass: MockRoutingService,
+        },
+        ProductReferencesService,
+      ],
+    });
+
+    // productCarouselService = TestBed.get(ProductReferencesService);
+  });
+
+  it('should inject ProductReferencesService', inject(
+    [ProductReferencesService],
+    (service: ProductReferencesService) => {
+      expect(service).toBeTruthy();
+    }
+  ));
+
+  it('should getReferenceType()', inject(
+    [ProductReferencesService],
```

Comment:
For the tests that come after should inject you should just use the `productCarouselService` defined in the beforeEach (in comments) rather than injecting the service each time. (Same for other test)